native-rendering/documents-designer/data-source.md

```diff
@@ -0,0 +1,106 @@
+---
+title: Documents data source
+---
+
+# Document data source
+
+During template design creation, document data can be accessed using JSONPath syntax. You can use both, the dot notation:
+
+```code
+$.shipment.tracking_number
+```
+or the bracket notation:
+
+```code
+$['shipment']['tracking_number']
+```
+## Notes:
+1. For now limited JSONPath syntax can be used. Notice that filter **expressions are not supported** yet.
+2. The leading `$.` or `.` or `@` is not optional. One of them have to be be used to differentiate variables and JSONPath syntax.
+### Exception:
+Special leading char `@` can be use as reference to current object but not in all cases. It depends of document type: e.g. for Package
+document type we can use `@package` reference to handle date from each package consecutively from multi packages shipment.
+
+3. Root element can be start from `$.` or `.`
+4. Relative nested object path have to be start from `.`
+If nested object is started from `$.` it means root element reference. In this case no relation will be consider.
+5. **Double dots ..property are NOT handled at all**
+6. Wildcard selecting all elements `[*]` can be used in an array only. Regardless of their indexes it means all items of the array.
+7. **Wildcard selecting all elements `.*` for object (means all properties) is NOT allowed. E.g. `shipment.*`**
+
+## JSONPath usage examples
+Below JSONPath examples are created for data source object structure:
+```code
+{
+	"shipment": {
+		"tracking_number": "123PL",
+		"packages": [
+			{
+				"weight": "12"
+			},
+			{
+				"weight": "02"
+			}
+		]
+	},
+	contract {
+	    "parameters":{
+			"Account.Number": "1234",
+			"IsBrexit" : "true"
+		}
+	}
+}
+```
+
+## Examples:
+
+### Properties
+
+```code
+$.shipment.tracking_number
+```
+
+## Arrays
+
+Can be reference with index or asterisk for each elements iteration
+```code
+$.shipment.packages[0].weight
+Sum($.shipment.packages[*].weight)
+$['shipment']['packages'][0]['weight']
+```
+
+## Dictionaries
+
+Generally dot and bracket notation can be used but in case when dictionary key contains dot in key name you have to used bracket notation only.
```

Comment:
```suggestion
Generally, a dot and bracket notation can be used, but in case a dictionary key contains a dot, you have to use bracket notation - see the third example.
```